WordPress/WordPressApi/WordPressComRestApi.swift

```diff
@@ -272,6 +247,163 @@ public class WordPressComRestApi: NSObject
         }
         return WordPressComRestApi.pathByAppendingPreferredLanguageLocale(path)
     }
+
+    private func URLForTemporaryFile() -> NSURL {
+        let fileName = "\(NSProcessInfo.processInfo().globallyUniqueString)_file.xmlrpc"
+        let fileURL = NSURL.fileURLWithPath(NSTemporaryDirectory()).URLByAppendingPathComponent(fileName)
+        return fileURL
+    }
+
+    private func multipartRequestWithURLString(urlString:String, parameters: [String:AnyObject], fileParts: [FilePart], encodedToFileURL fileURL:NSURL) -> NSURLRequest? {
+        //let urlStringWithLocale = appendLocaleIfNeeded(urlString)
+        guard let baseURL = NSURL(string: WordPressComRestApi.apiBaseURLString),
+              let url = NSURL(string:urlString, relativeToURL:baseURL)
+        else {
+            return nil
+        }
+        let request = NSMutableURLRequest(URL: url)
+        request.HTTPMethod = "POST"
+        let boundary = NSUUID().UUIDString
+        request.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField:"Content-Type")
+        let body = NSMutableData()
+
+        for (key, obj) in parameters {
+            if let text = obj as? String {
```

Comment:
(If) we can't switch over to [String: String], i think these two lines can be expressed as:

```
for (key, text) in parameters where text is String
```